File: pdlearning/src/web-apps/src/libraries/domain-api/src/content/models/video-comment.model.ts
export interface IVideoComment {
  id: string;
  userId: string;
  objectId: string;
  sourceType: VideoCommentSourceType;
  originalObjectId: string;
  content: string;
  videoId: string;
  videoTime: number;
  createdDate: Date;
  changedDate?: Date;
}

export class VideoComment implements IVideoComment {
  public id: string;
  public userId: string;
  public objectId: string;
  public sourceType: VideoCommentSourceType;
  public originalObjectId: string;
  public content: string;
  public videoId: string;
  public videoTime: number;
  public createdDate: Date;
  public changedDate?: Date;
  constructor(data?: IVideoComment) {
    if (data == null) {
      return;
    }
    this.id = data.id;
    this.userId = data.userId;
    this.objectId = data.objectId;
    this.sourceType = data.sourceType;
    this.originalObjectId = data.originalObjectId;
    this.content = data.content;
    this.videoId = data.videoId;
    this.videoTime = data.videoTime;
    this.createdDate = new Date(data.createdDate);
    this.changedDate = data.changedDate ? new Date(data.changedDate) : null;
  }
}

export enum VideoCommentSourceType {
  CCPM = 'CCPM',
  LMM = 'LMM',
  CSL = 'CSL'
}

File: pdlearning/src/microservices/content-microservice/Api/Application/Models/VideoCommentModel.cs
using System;
using Microservice.Content.Domain.ValueObject;
using VideoCommentEntity = Microservice.Content.Domain.Entities.VideoComment;

namespace Microservice.Content.Application.Models
{
    public class VideoCommentModel
    {
        public VideoCommentModel()
        {
        }

        public VideoCommentModel(VideoCommentEntity comment)
        {
            Id = comment.Id;
            UserId = comment.UserId;
            ObjectId = comment.ObjectId ?? Guid.Empty;
            OriginalObjectId = comment.OriginalObjectId ?? Guid.Empty;
            SourceType = comment.SourceType;
            Content = comment.Content;
            VideoId = comment.VideoId;
            VideoTime = comment.VideoTime;
            Note = comment.Note;
            CreatedDate = comment.CreatedDate;
            ChangedDate = comment.ChangedDate;
        }

        public Guid Id { get; set; }

        public Guid UserId { get; set; }

        public Guid ObjectId { get; set; }

        public VideoSourceType SourceType { get; set; }

        public Guid OriginalObjectId { get; set; }

        public string Content { get; set; }

        public Guid VideoId { get; set; }

        public int VideoTime { get; set; }

        public string Note { get; set; }

        public DateTime CreatedDate { get; set; }

        public DateTime? ChangedDate { get; set; }
    }
}
